src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -215,17 +215,59 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Default(byte* pBuffer, n
             goto Finish;
         }
 
-        private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuint bufferLength)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static ulong GetMaskOfMostSignificantBits(Vector128<byte> value)
+        {
+            Debug.Assert(BitConverter.IsLittleEndian);
+            if (!AdvSimd.Arm64.IsSupported)
+            {
+                throw new PlatformNotSupportedException();
+            }
+
+            // extractedBits[i] = (value[i] & 0x80 == 0x80) & (1 << (12 * (i % 2)))
+            Vector128<byte> s_bitmask = Vector128.Create((ushort)0x1001).AsByte();
+            Vector128<byte> mostSignificantBitIsSet = AdvSimd.ShiftRightArithmetic(value.AsSByte(), 7).AsByte();
+            Vector128<byte> extractedBits = AdvSimd.And(mostSignificantBitIsSet, s_bitmask);
+
+            extractedBits = AdvSimd.Arm64.AddPairwise(extractedBits, extractedBits);
+            extractedBits = AdvSimd.Arm64.AddPairwise(extractedBits, extractedBits);
+            extractedBits = AdvSimd.Arm64.AddPairwise(extractedBits, extractedBits);
+            return extractedBits.AsUInt64().ToScalar();
```

Comment:
To match `Sse2.MoveMask` behaviour, this should only return the first 16 bits of the vector. 
```suggestion
            return extractedBits.AsUInt16().ToScalar();
```

Depending on how you use the mask, this may not have any impact, but as it stands bits will repeated 4 times in the returned value.